### ISSUE TITLE
Additional Directory Keywords

### ISSUE THREAD

**Issue by rstoneback** [pysat/pysat#183]
Support more complicated directory structures with a file_directory keyword, or something like that.

file_format for file name formatting
file_directory for additional directory formatting outside that standard pysat structure

Must be compatible with existing directory structure keyword that accounts for things that name, tag, and sat_id.

To close:
- [x] Need to close #309 
- [x] Need to close #123 
- [x] Add more custom directory options

**Comment by jklenzing** [comment]
Also add possible breakdowns by year/month/day for extremely large datasets (eg,ground-based observatories)

**Comment by jklenzing** [comment]
Proposing directory changes move to 3.0.0

**Comment by rstoneback** [comment]
yeah, I agree. I came here to say that and I'm behind the times :)

This requires an API change and should be v3.0 or later.

**Comment by jklenzing** [comment]
Link to existing issues #123 and #58

**Comment by rstoneback** [comment]
There is a working version of this in #637. I didn't add support for a keyword as originally written. I followed a style similar to the matplotlib rcparams file, as suggested over in the params issue #309, in that multiple directories may be specified as pysat data directories, and the first one that pysat finds with files is *the* directory. pysat stops looking after it finds something. 

A consequence of this is that currently instantiated Instruments no longer look to a newly set directory, they continue on with the directory they started with, until a `.refresh()` is called. This is probably a good thing. I also added `update_files` as a stored pysat parameters, with a default of True, so that all newly instantiated Instruments will query the filesystem and be responsive to latest settings.
